### ISSUE TITLE
/usr/bin/python3 cannot be found by default test harness

### ISSUE THREAD

**Issue by sharwell** [antlr/antlr4-python3#41]
The default installation location of Python 3.4 on a fresh Ubuntu installation is /usr/bin/python3. This location should be detected and used by the test suite without any additional configuration.

**Comment by ericvergnaud** [comment]
feel free to propose a fix
I don't have a fresh Ubuntu installation to test against

**Comment by parrt** [comment]
```java
java.lang.RuntimeException: Could not locate Python3.4
	at org.antlr.v4.test.runtime.python.BasePythonTest.locateTool(BasePythonTest.java:561)
	at org.antlr.v4.test.runtime.python.BasePythonTest.locatePython(BasePythonTest.java:568)
	at org.antlr.v4.test.runtime.python.BasePythonTest.execModule(BasePythonTest.java:525)
	at org.antlr.v4.test.runtime.python.BasePythonTest.execRecognizer(BasePythonTest.java:521)
	at org.antlr.v4.test.runtime.python.BasePythonTest.execParser(BasePythonTest.java:454)
	at org.antlr.v4.test.runtime.python.BasePythonTest.execParser(BasePythonTest.java:425)
	at org.antlr.v4.test.runtime.python3.TestParserExec.testPredicatedIfIfElse(TestParserExec.java:128)
...
```

The issue is that it's looking for Python3.4 not `python3` right?

**Comment by ericvergnaud** [comment]
what's missing is python installing links
you need to update BasePythonTest here:
		String[] roots = { "/usr/bin/", "/usr/local/bin/" };

**Comment by parrt** [comment]
Hmm...actually,It looks proper:

```
parrt@hrn23502:~/antlr/code/antlr4$ which python3
/usr/bin/python3
```

 should there be an alias to python3.4? Note that the error message seems to be looking for `Python3.4` capitalized.

**Comment by sharwell** [comment]
Originally my thought is we could update the test to look for `python3` if `python3.4` isn't found. However, it seems like that might not be as trivial as looking for `python3` *instead* of `python3.4`.

:thought_balloon: It seems like we could update [this line](https://github.com/antlr/antlr4-python3/blob/214a4b7f52a1755f93fd9e3a7bf8bc3edd72dc51/tool/test/org/antlr/v4/test/runtime/python3/BasePython3Test.java#L44) to be `python3` instead of `python3.4`.

@ericvergnaud Do you see the potential for a negative impact of looking for `python3` instead of `python3.4`? Between looking for two executables and changing the executable, which one has more downsides?

**Comment by ericvergnaud** [comment]
there is a possibility that the issue comes from Ubuntu being case sensitive
please pull latest Python to check
I am not keen on managing tests failures if python3 is not 3.4
there have been major upgrades since 3.0 which should prevent the tests from passing and generate pointless support

**Comment by sharwell** [comment]
Fixed in 214a4b7f52a1755f93fd9e3a7bf8bc3edd72dc51. In my previous comment, I didn't realize a change was actually made between the time I reported this issue and the time I made that comment. Prior to 214a4b7f52a1755f93fd9e3a7bf8bc3edd72dc51, the code was looking for a `Python3.4` executable (which does not exist on the system I'm using), but the new code looks for `python3.4` (which does exist).
